provider/provider.go

```diff
@@ -176,6 +182,20 @@ func (p *Provider) FetchResources(ctx context.Context, request *cqproto.FetchRes
 	return g.Wait()
 }
 
+func (p *Provider) interpolateAllResources(requestedResources []string) []string {
+	if len(requestedResources) != 1 {
+		return requestedResources
+	}
+	if requestedResources[0] != "*" {
+		return requestedResources
+	}
+	allResources := make([]string, 0)
```

Comment:
```suggestion
	allResources := make([]string, 0, len(p.ResourceMap))
```
(could also use a counter variable and do `allResources[i] = k; i++` and get rid of the `append`...)

Reply:
added the cap, I like the append feels more cleaner than adding a counter.